docs/standard/commandline/beta5.md

```diff
@@ -0,0 +1,330 @@
+| `System.CommandLine.Argument.AddCompletions`            | `System.CommandLine.Argument.CompletionSources`             |
+| `System.CommandLine.Command.AddAlias`                   | `System.CommandLine.Command.Aliases`             |
+| `System.CommandLine.Option.AddAlias`                    | `System.CommandLine.Option.Aliases`             |
+
+The `RemoveAlias` and `HasAlias` methods were also removed, as the `Aliases` property is now a mutable collection. You can use the `Remove` method to remove an alias from the collection, and `Contains` method to check if an alias exists.
+
+## Names and aliases
+
+Before beta5, there was no clear separation between the name and [aliases](syntax.md#aliases) of a symbol. When `name` was not provided for the `Option<T>` constructor, the symbol reported its name as the longest alias with prefixes like `--`, `-`, or `/` removed. That was confusing.
+
+Moreover, to get the parsed value, users had to store a reference to an option or an argument and then use it to get the value from `ParseResult`.
+
+To promote simplicity and explicitness, we decided to make the name of a symbol a mandatory parameter for every symbol constructor (including `Argument<T>`). We have also separated the concept of a name and aliases; now aliases are just aliases and do not include the name of the symbol. Of course, they are optional. As a result, we made the following changes:
+
+- `name` is now a mandatory argument for every public constructor of `Argument<T>`, `Option<T>`, and `Command`. In the case of `Argument<T>`, it is not used for parsing, but to generate the help and completions text. In the case of `Option<T>` and `Command`, it is used to identify the symbol during parsing and also for help and completions.
+- The `Symbol.Name` property is no longer `virtual`; it's now read-only and returns the name as it was provided when the symbol was created. Because of that, `Symbol.DefaultName` was removed and `Option.Name` no longer removes the `--`, `-`, or `/` or any other prefix from the longest alias.
+- The `Aliases` property exposed by `Option` and `Command` is now a mutable collection. This collection no longer includes the name of the symbol.
+- `System.CommandLine.Parsing.IdentifierSymbol` was removed (it was a base type for both `Command` and `Option`).
+
+Having the name always present allows for [getting the parsed value by name](parse-and-invoke.md#getvalue):
+
+```csharp
+RootCommand command = new("The description.")
+{
+    new Option<int>("--number")
+};
+
+ParseResult parseResult = command.Parse(args);
+int number = parseResult.GetValue<int>("--number");
+```
+
+### Creating options with aliases
+
+In the past, `Option<T>` was exposing plenty of constructors, some of which were accepting the name or not. Since the name is now mandatory and we expect aliases to be frequetly provided for `Option<T>`, we provide only a single constrtor. It accepts the name and a `params` array of aliases.
```

Comment:
```suggestion
In the past, `Option<T>` exposed many constructors, some of which accepted the name. Since the name is now mandatory and we expect aliases to be frequently provided for `Option<T>`, we provide only a single constructor. It accepts the name and a `params` array of aliases.
```